rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,150 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
 1. Resolve peer's node DID to its corresponding DID document.
 2. Assert that one of the `NutsComm` endpoint's host matches (one of) the `dNSName` SANs in the peer's TLS client certificate.
 
-## X. Private Transactions
\ No newline at end of file
+## 4. Set reconciliation protocol
+
+### 4.1 Data requirements
+
+The protocol requires an XOR value and IBLT structure to be sent in a message. 
+These are to be calculated over different transaction ranges based on LC value.
+Both data structures have associative properties allowing for optimized data storage.
+
+#### 4.1.1 XOR
+
+The exclusive-or (XOR) value is calculated over all transactions reference hashes within a range.
+The length of the XOR value is therefore the same as the transaction reference length (SHA256: 32 bytes).
+The XOR value is used to quickly determine if two nodes have processed the exact same set of transactions.
+
+#### 4.1.2 Invertible Bloom Lookup Table
+
+The IBLT is a data structure capable of finding differences in sets.
+The performance of this data structure is not impacted by the number of entries, but only by the size of the set difference.
+This allows two nodes to compare the entire set of transactions with a relative small data structure.
+
+An IBLT has several parameters that define its characteristics. When comparing two IBLTs, these parameters MUST be the same.
+An overview of the parameters:
+
+* **#buckets**: defines the size of the IBLT, much like the size of a bloom filter. The larger the size, the bigger the set difference can be. 
+* **Hk**: each key is hashed to select the buckets the key is inserted to. The hash function is applied on the previous output to create extra hash values. This continues until k different values are found. The hash function, its seed and the value for k MUST all be the same for the entire network.
```

Comment:
detail: the hashing function produces numbers much larger than the number of buckets. Since 2 different hashes can map to the same bucket, we need to keep hashing until we find k different buckets not values.

Reply:
correct, as described in the paper.
If a hash function is uniform, doesn't that automatically mean that hashing twice is uniform?

Reply:
Yes, it should be uniform on random inputs. I just don't know if the hash-RNG provides any guarantees about how many numbers are generated before it repeats itself. This period could provide less than the k buckets needed to put the key/TxRef in the IBLT.